packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -647,23 +648,19 @@ export class ConnectionManager implements IConnectionManager {
 
 				const waitStartTime = performance.now();
 				const retryDelayFromError = getRetryDelayFromError(origError);
-				// If the error told us to wait, then we wait for that time. If the error didn't tell us to wait,
-				// let's still wait a little bit before retrying. We skip this delay if we're confident we're offline,
+				// If the error told us to wait or browser signals us that we are offline, then calculate the time we
+				// want to wait for before retrying. then we wait for that time. If the error didn't tell us to wait,
+				// let's still wait a little bit before retrying. We can skip this delay if we're confident we're offline,
 				// because we probably just need to wait to come back online. But we never have strong signal of being
 				// offline, so we at least wait for sometime.
 				if (retryDelayFromError !== undefined || globalThis.navigator?.onLine !== false) {
-					delayMs = Math.max(retryDelayFromError ?? 0, delayMs);
-					// Raise event in case the delay was there.
-					this.props.reconnectionDelayHandler(delayMs, origError);
-					await new Promise<void>((resolve) => {
-						setTimeout(resolve, delayMs);
-					});
-					delayMs = Math.min(delayMs * 2, calculateMaxWaitTime(origError));
-				} else {
-					await new Promise<void>((resolve) => {
-						setTimeout(resolve, 200);
-					});
+					delayMs = calculateMaxWaitTime(delayMs, origError);
 				}
+				// Raise event in case the delay was there.
+				this.props.reconnectionDelayHandler(delayMs, origError);
+				await new Promise<void>((resolve) => {
+					setTimeout(resolve, delayMs);
+				});
```

Comment:
Is that's correct? Old code would increase delay up to 8 seconds in offline, but new code runs with same frequency (which is half of starting frequency). By offline I mean case where browser does not know that we are offline - we simply can't reach endpoint.

Reply:
Old code would have only increased when "globalThis.navigator?.onLine !== false" depending upon whether endpoint was reached or not. and waited for 200 ms(0 sec before previous change). Now we are waiting for fix delayMs now.